app/app.go

```diff
@@ -964,12 +965,17 @@ func (app *App) ProcessBlockSynchronous(ctx sdk.Context, txs [][]byte) []*abci.E
 }
 
 // Returns a mapping of the accessOperation to the channels
-func getChannelsFromSignalMapping(signalMapping acltypes.MessageCompletionSignalMapping) sdkacltypes.MessageAccessOpsChannelMapping {
+func GetChannelsFromSignalMapping(signalMapping acltypes.MessageCompletionSignalMapping) sdkacltypes.MessageAccessOpsChannelMapping {
 	channelsMapping := make(sdkacltypes.MessageAccessOpsChannelMapping)
 	for messageIndex, accessOperationsToSignal := range signalMapping {
 		for accessOperation, completionSignals := range accessOperationsToSignal {
 			var channels []chan interface{}
-			channelsMapping[messageIndex] = make(sdkacltypes.AccessOpsChannelMapping)
+			if val, ok := channelsMapping[messageIndex]; ok {
+				channels = val[accessOperation]
```

Comment:
why do we need to get the channels here? since we're iterating over accessOperation, shouldnt we be starting with a fresh slice each iteration? (aka there shouldnt be an entry for `val[accessOperation]` right?